### ISSUE TITLE
watch seems to be monitoring changes inside my .git folder

### ISSUE THREAD

**Issue by callumlocke** [yeoman/generator-webapp_DEPRECATED#298]
I think this is a recent change that is probably a bug...

When I have `grunt serve` running in one terminal, and then do a `git commit` in another terminal, I see loads of this sort of thing:

```
>> File ".git/objects/0c" added.
>> File ".git/objects/2e" added.
>> File ".git/objects/57" added.
>> File ".git/objects/a9" added.
>> File ".git/objects/c2" added.
>> File ".git/objects/c5" added.
>> File ".git/objects/f6" added.

OK
>> File ".git/objects/5e" added.
>> File ".git/objects/98" added.
>> File ".git/objects/e0" added.

OK
```

This isn't causing any particular problem for me, but I'm guessing it's a big waste of memory if something is watching the `.git` directory recursively.

(Using latest master of generator-webapp btw.)

**Comment by addyosmani** [comment]
What happens if you explicitly exclude `.git` from the watch task? Could you see if it still gets triggered?

**Comment by hemanth** [comment]
Watch [config](https://github.com/yeoman/generator-webapp/blob/master/app/templates/Gruntfile.js#L29) is watching for:

``` sh
Watching app\scripts\main.js for changes.
Watching test\spec\test.js for changes.
Watching Gruntfile.js for changes.
Watching .tmp for changes.
Watching app for changes.
Watching node_modules for changes.
Watching test for changes.
Watching app\styles\main.css for changes.
Watching app\404.html for changes.
Watching app\bower_components for changes.
Watching app\images for changes.
Watching app\scripts for changes.
Watching app\styles for changes.
Watching app\index.html for changes.
Watching .tmp\styles\main.css for changes.
```

Adding `'!.**/*'` does help, but dot files are not picked unless `{options: { dot: true }}`

![](http://www.reactiongifs.com/r/spin.gif)

**Comment by eddiemonge** [comment]
@callumlocke have you changed anything in your gruntfile?

**Comment by callumlocke** [comment]
Hey, sorry I didn't get back to you. No, I didn't change anything in my Gruntfile, this happened immediately with a new project.

I just managed to replicate it with a new project, and I've realised it happens only if your `.git` folder gets created while you're **already** watching, ie:
1. `yo webapp`
2. `grunt serve`
3. in another terminal: `git init`

If you do things in that order, you see `>> File ".git" added.` in your grunt serve terminal (and a bunch of other 'added' messages, and loads more when you commit).

Possibly a bug in grunt-contrib-watch or gaze?
